x/data/server/msg_anchor.go

```diff
@@ -37,14 +37,33 @@ func (s serverImpl) anchorAndGetIRI(ctx context.Context, ch ToIRI) (iri string,
 		return "", nil, nil, err
 	}
 
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
-	store := sdkCtx.KVStore(s.storeKey)
-	id = s.iriIDTable.GetOrCreateID(store, []byte(iri))
+	dataId := &api.DataID{Iri: ""}
+
+	for collisions := 0; dataId.Iri != iri; collisions++ {
```

Comment:
would be good to at least break this out into a standalone function

Reply:
https://github.com/regen-network/regen-ledger/pull/970/commits/16b101c0f8b09fb5f8db7232707d2c8998d07781